packages/view-types/statistical-plots/src/ExpressionHistogramSubscriber.js

```diff
@@ -67,26 +89,54 @@ export function ExpressionHistogramSubscriber(props) {
   // From the expression matrix and the list of selected genes,
   // generate the array of data points for the histogram.
   const data = useMemo(() => {
+    console.log("data is recomputed");
     if (firstGeneSelected && obsFeatureMatrix && expressionData) {
+      console.log("gene is defined");
+      console.log("** expression data: ", expressionData[0]);
       // Create new cellColors map based on the selected gene.
-      return Array.from(expressionData[0]).map((_, index) => {
+      const newData = Array.from(expressionData[0]).map((_, index) => {
         const value = expressionData[0][index];
         const normValue = value * 100 / 255;
-        return { value: normValue, gene: firstGeneSelected };
+        const newItem = { value: normValue, gene: firstGeneSelected };
+        return newItem;
       });
+      console.log("I want to return this data:", newData);
+      setDataOnSelect(newData);
+      return newData;
     }
     if (obsFeatureMatrix) {
+      console.log("gene is not defined");
+      console.log("** obsFeatureMatrix: ", obsFeatureMatrix.data);
       const numGenes = featureIndex.length;
-      return obsIndex.map((cellId, cellIndex) => {
+      const newData = obsIndex.map((cellId, cellIndex) => {
         const values = obsFeatureMatrix.data
           .subarray(cellIndex * numGenes, (cellIndex + 1) * numGenes);
         const sumValue = sum(values) * 100 / 255;
-        return { value: sumValue, gene: null };
+        const newItem = { value: sumValue, gene: null, cellId };
+        return newItem;
       });
+      setDataOnSelect(newData);
+      return newData;
     }
     return null;
   }, [obsIndex, featureIndex, obsFeatureMatrix, firstGeneSelected, expressionData]);
 
+  const onSelect = useCallback((value) => {
+
+    const geneName = firstGeneSelected ? [firstGeneSelected, 'values'].join(' ') : 'transcript count';
+
+    const selectedCellIds = getObsInfoFromDataWithinRange(value, dataOnSelectRef.current);
+    setObsSelection(
+      selectedCellIds, additionalCellSets, cellSetColor,
+      setCellSetSelection, setAdditionalCellSets, setCellSetColor,
+      setCellColorEncoding,
+      '',
+      `Selection based on ${geneName} in range [${value[0].toFixed()}, ${value[1].toFixed()}].`,
```

Comment:
Good point, I did not think about this. It is possible for two sets with the same name to be created, and different number of cell ids (it is because the lower and upper bound in range are floats, and in the name we round them). In this case, I wouldn't silently skip the creation of a new selection, because the user might think that it is a bug on our side. I recommend that we either raise an error and display it as a popup, saying something like "Couldn't create new cell selection, because a cell set with the same ranges already exists". Or we should make sure that every cell selection has unique name.
Seems to me that the latter option is more consistent with the behaviour of Vitessce and I adjusted the code to support it. I am adding the ability to specify a "suffix" in the `setObsSelection` function. When suffix is supplied, the new selection name is equal to prefix + the increment + suffix. I also use `toFixed(1)` when displaying the upper and lower bounds. It seems to me that this looks ok. Here is a screenshot, what do you think?
<img width="701" alt="Screenshot 2023-05-10 at 14 11 39" src="https://github.com/vitessce/vitessce/assets/65414651/ef16a1eb-8d9c-4612-bc6e-b0d364507fee">

Reply:
I like this!